File: Single and multiple Inheritance.py
class Engine():
    EngineName = ''

    def __init__(self,en) -> None:
        self.EngineName = en

    def engineCompany():
        print('Chinese COmpany')

class Window():
    bulletproof = ''
    def __init__(self,bp) -> None:
        self.bulletproof = bp


#multiple Inheritance
class Car(Engine , Window):
    model = ''
    def __init__(self, model,en,bp) -> None:
        self.model = model
        Engine.__init__(self,en)
        Window.__init__(self,bp)

myCar = Car('BMW', 'Castor' , 'YES')

print(myCar.EngineName)

#Single Inheritance

class Bike(Engine):
    model = ''
    def __init__(self, model,en) -> None:
        self.model = model
        Engine.__init__(self,en)

myBike = Bike('Pulser', 'Chinese')

print(myBike.model)
        

File: Single and Multiple Inheritance.cpp
#include <bits/stdc++.h>
using namespace std;

class Engine {
public:
    string EngineName;

    Engine(string en) {
        this->EngineName = en;
    }

        void engineCompany() {
        cout << "Chinese Company" << endl;
    }
};

class Window {
public:
    string bulletproof;

    Window(string bp) {
        this->bulletproof = bp;
    }
};
//Multiple Inheritance
class Car : public Engine, public Window {
public:
    string model;

    Car(string model, string en, string bp) : Engine(en), Window(bp) {
        this->model = model;
    }
};
//single inhertance
class Bike : public Engine {
public:
    string model;

    Bike(string model, string en) : Engine(en) {
        this->model = model;
    }
};

int main() {
    Car myCar("Audi", "Motoryy", "No");
    cout << myCar.EngineName << endl;

    Bike myBike("FZ", "Chinese");
    cout << myBike.model << endl;

    return 0;
}
